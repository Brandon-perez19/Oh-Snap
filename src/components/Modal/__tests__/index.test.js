import React from 'react';
import { render, cleanup, fireEvent} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Modal from '..';

const MockToggleModal = jest.fn();
const currentPhoto = {
    name: 'Park Bench',
    category: 'landscape',
    description:'Lorem ipsum',
    index: 1
};

afterEach(cleanup);

describe('Modal component', () => {
    it('renders', () => {
        render(<Modal currentPhoto={currentPhoto} onClose={MockToggleModal} />)
    });

    it('matches snapshot DOM node structure', () => {
        const {asFragment} = render(<Modal currentPhoto={currentPhoto} onClose={MockToggleModal} />)
        expect(asFragment()).toMatchSnapshot();
    })
});

describe('Click Event', () => {
    it('calls onClose handler', () => {
        const { getByText } = render(<Modal
            onClose={MockToggleModal}
            currentPhoto={currentPhoto}
        />);
        fireEvent.click(getByText('Close this modal'));
        expect(MockToggleModal).toHaveBeenCalledTimes(1);
    })
})