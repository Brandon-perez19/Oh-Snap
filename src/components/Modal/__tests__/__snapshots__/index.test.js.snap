// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div
    class="modalBackdrop"
  >
    <div
      class="modalContainer"
    >
      <h3
        class="modalTitle"
      >
         Park Bench 
      </h3>
      <img
        alt="current category"
        src="1.jpg"
      />
      <p>
        Lorem ipsum
      </p>
      <button
        type="button"
      >
        Close this modal
      </button>
    </div>
  </div>
</DocumentFragment>
`;
